x-pack/plugins/ml/public/application/datavisualizer/stats_table/data_visualizer_stats_table.tsx

```diff
@@ -19,61 +19,62 @@ import {
 } from '@elastic/eui';
 import { i18n } from '@kbn/i18n';
 import { EuiTableComputedColumnType } from '@elastic/eui/src/components/basic_table/table_types';
+import { ML_JOB_FIELD_TYPES } from '../../../../common/constants/field_types';
 import { FieldTypeIcon } from '../../components/field_type_icon';
-import { FieldVisConfig } from '../index_based/common';
-import { DataVisualizerFieldExpandedRow } from './expanded_row';
 import { DocumentStat } from './components/field_data_row/document_stats';
 import { DistinctValues } from './components/field_data_row/distinct_values';
-import { NumberContentPreview } from './components/field_data_row/number_content_preview';
-import { DataVisualizerIndexBasedAppState } from '../../../../common/types/ml_url_generator';
+import { IndexBasedNumberContentPreview } from './components/field_data_row/number_content_preview';
+import {
+  DataVisualizerFileBasedAppState,
+  DataVisualizerIndexBasedAppState,
+} from '../../../../common/types/ml_url_generator';
 import { useTableSettings } from '../../data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings';
 import { TopValuesPreview } from './components/field_data_row/top_values_preview';
 import type { MlJobFieldType } from '../../../../common/types/field_types';
-const FIELD_NAME = 'fieldName';
+import {
+  FieldVisConfig,
+  FileBasedFieldVisConfig,
+  isIndexBasedFieldVisConfig,
+} from './types/field_vis_config';
+import { FileBasedNumberContentPreview } from '../file_based/components/field_data_row';
+import { BooleanContentPreview } from './components/field_data_row';
 
-interface DataVisualizerDataGrid {
-  items: FieldVisConfig[];
-  pageState: DataVisualizerIndexBasedAppState;
-  updatePageState: (update: Partial<DataVisualizerIndexBasedAppState>) => void;
-}
+const FIELD_NAME = 'fieldName';
 
 export type ItemIdToExpandedRowMap = Record<string, JSX.Element>;
 
-function getItemIdToExpandedRowMap(
-  itemIds: string[],
-  items: FieldVisConfig[]
-): ItemIdToExpandedRowMap {
-  return itemIds.reduce((m: ItemIdToExpandedRowMap, fieldName: string) => {
-    const item = items.find((fieldVisConfig) => fieldVisConfig[FIELD_NAME] === fieldName);
-    if (item !== undefined) {
-      m[fieldName] = <DataVisualizerFieldExpandedRow item={item} />;
-    }
-    return m;
-  }, {} as ItemIdToExpandedRowMap);
+type DataVisualizerTableItem = FieldVisConfig | FileBasedFieldVisConfig;
+interface DataVisualizerTableProps<T> {
+  items: T[];
+  pageState: DataVisualizerIndexBasedAppState | DataVisualizerFileBasedAppState;
+  updatePageState: (update: any) => void;
```

Comment:
please avoid using `any` if possible. `update` can be `Partial<DataVisualizerIndexBasedAppState  | DataVisualizerFileBasedAppState>` I presume

Reply:
Updated here https://github.com/elastic/kibana/pull/87598/commits/e6f1c5f39d01540e0a24875062722a36360390f1